### PR TITLE
Remove 3.8 rolling upgrade checks

### PR DESCRIPTION
**Pull request by mmedenjak**
Also added missing RU_COMPAT comment.

### CODE REVIEW

**Comment by mmedenjak** [comment]
run-lab-run

**Comment by mmedenjak** [comment]
run-lab-run

File: hazelcast/src/test/java/com/hazelcast/concurrent/lock/ObjectNamespaceSerializationHelperTest.java [hunk 1]
```diff
@@ -84,13 +79,19 @@ public void readNamespaceCompatibly() throws Exception {
     private ObjectNamespace serialiseDeserialise(ObjectNamespace ns, Version streamVersion) throws IOException {
         final byte[] bytes = serialize(ns, streamVersion);
         final BufferObjectDataInput input = ss.createObjectDataInput(bytes);
-        return ObjectNamespaceSerializationHelper.readNamespaceCompatibly(input);
```
Comment (mdogan) [inline hazelcast/src/test/java/com/hazelcast/concurrent/lock/ObjectNamespaceSerializationHelperTest.java]: `ObjectNamespaceSerializationHelper` might still be useful. I see the same write/read logic used in several places. OR we can remove `DefaultObjectNamespace` completely.

File: hazelcast/src/main/java/com/hazelcast/spi/impl/eventservice/impl/operations/AbstractRegistrationOperation.java [hunk 2]
```diff
@@ -53,11 +52,6 @@ public final void run() throws Exception {
     protected abstract void runInternal() throws Exception;
 
     private void checkMemberListVersion() {
-        if (memberListVersion == -1) {
```
Comment (mdogan) [inline hazelcast/src/main/java/com/hazelcast/spi/impl/eventservice/impl/operations/AbstractRegistrationOperation.java]: This is already removed by https://github.com/hazelcast/hazelcast/commit/982fd3bcc1bf8bf528edfb84b463f841195c1408

Comment (mmedenjak) [reply hazelcast/src/main/java/com/hazelcast/spi/impl/eventservice/impl/operations/AbstractRegistrationOperation.java]: Rebased.

File: hazelcast/src/test/java/com/hazelcast/concurrent/lock/ObjectNamespaceSerializationHelperTest.java (hunk 1 above)
Comment (mmedenjak) [reply hazelcast/src/test/java/com/hazelcast/concurrent/lock/ObjectNamespaceSerializationHelperTest.java]: Left the `ObjectNamespaceSerializationHelper` class. Can you check now?

**Pull request merged**